libs/ngxtension/computed-async/src/computed-async.ts

```diff
@@ -72,46 +96,157 @@ interface ComputedAsyncOptions<T> extends CreateComputedOptions<T> {
+): Signal<T | undefined>;
+
+/*
+ * Promise Types
+ */
+
+// Options with `undefined` initial value -> result includes `undefined`.
+export function computedAsync<T>(
+	computation: (previousValue?: T | undefined) => Promise<T> | T | undefined,
+	options: OptionsWithOptionalInitialValue<T>,
+): Signal<T | undefined>;
+
+// Options with `null` initial value -> `null`.
+export function computedAsync<T>(
+	computation: PromiseComputation<T>,
+	options: { initialValue?: null } & ComputedAsyncOptions<T>,
+): Signal<T | null>;
+
+// Options with a more specific initial value type.
+export function computedAsync<T>(
+	computation: PromiseComputation<T>,
+	options: OptionsWithInitialValue<T>,
+): Signal<T>;
+
+// Options with `requireSync` -> never.
+export function computedAsync<T>(
+	computation: (previousValue?: T | undefined) => Promise<T>,
+	options: OptionsWithOptionalInitialValue<T> & {
+		/**
+		 * @throws Because the promise will not resolve synchronously.
+		 */
+		requireSync: true;
+	},
+): never;
+
+/*
+ * Observable Types
+ */
+
+// Options with `undefined` initial value and no `requireSync` -> `undefined`.
 export function computedAsync<T>(
-	computation: (previousValue?: T | undefined) => ComputationResult<T>,
-): Signal<T|undefined>
+	computation: (previousValue?: T | undefined) => Observable<T> | T | undefined,
+	options: {
+		initialValue?: undefined;
+		requireSync?: false;
+	} & ComputedAsyncOptions<T>,
+): Signal<T | undefined>;
+
+// Options with `null` initial value -> `null`.
 export function computedAsync<T>(
-	computation: (previousValue?: T | undefined) => ComputationResult<T>,
-  options: {initialValue:undefined} & ComputedAsyncOptions<T>,
-): Signal<T|undefined>
+	computation: ObservableComputation<T>,
+	options: {
+		initialValue?: null;
+		requireSync?: false;
+	} & ComputedAsyncOptions<T>,
+): Signal<T | null>;
+
+// Options with `undefined` initial value and `requireSync` -> strict result type.
 export function computedAsync<T>(
-	computation: (previousValue?: T | undefined) => ComputationResult<T>,
-	options: ComputedAsyncOptions<T>,
-): Signal<T>
+	computation: ObservableComputation<T>,
+	options: OptionsWithRequireSync<T> & { initialValue?: undefined },
+): Signal<T>;
+
+// Options with `T` initial value and `requireSync` -> strict result type.
 export function computedAsync<T>(
-	computation: (previousValue?: T | undefined) => ComputationResult<T>,
-	options?: ComputedAsyncOptions<T>,
-): Signal<T|undefined> {
+	computation: ObservableComputation<T>,
+	options: OptionsWithRequireSync<T> & { initialValue: T },
+): Signal<T>;
+
+// Options with `T` initial value and no `requireSync` -> strict result type.
+export function computedAsync<T>(
+	computation: ObservableComputation<T>,
+	options: OptionsWithInitialValue<T>,
+): Signal<T>;
+
+export function computedAsync<T>(
+	computation: (
+		previousValue?: T | undefined,
+	) => Promise<T> | Observable<T> | T | undefined,
+	options?: any,
```

Comment:
nit: I would personally set this to be an empty object.
```suggestion
	options: any = {}
```